lib/src/widget/built_in/simple/simple.dart

```diff
@@ -77,32 +93,43 @@ class SimpleToastWidget extends StatelessWidget {
     required BorderSide borderSide,
     required BorderRadiusGeometry borderRadius,
     required bool applyBlurEffect,
+    required bool showCloseButton,
   }) {
-    Widget body = Container(
+    Widget body;
+
+    if (showCloseButton) {
+      body = Row(
+        children: [
+          Expanded(
```

Comment:
If you want to have a widget which allocate size only base on its content you can use Flexible widget 

Check out this example:

``` dart 

    Row(
      mainAxisSize: MainAxisSize.min,
      children: [
        Flexible(
          fit: FlexFit.loose,
          child: widget.title ?? const SizedBox(),
        ),
        ToastCloseButton(
          showCloseButton: showCloseButton,
          onCloseTap: widget.onCloseTap,
          icon: widget.defaultStyle.closeIcon(context),
          iconColor: widget.foregroundColor?.withOpacity(.3) ??
              widget.defaultStyle.closeIconColor(context),
        ),
      ],
    )
```